python/plugins/processing/algs/qgis/RasterCalculator.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    RasterLayerCalculator.py
+    ---------------------
+    Date                 : November 2016
+    Copyright            : (C) 2016 by Victor Olaya
+    Email                : volayaf at gmail dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+from processing.modeler.ModelerAlgorithm import ValueFromInput, ValueFromOutput
+import os
+
+
+__author__ = 'Victor Olaya'
+__date__ = 'November 2016'
+__copyright__ = '(C) 2016, Victor Olaya'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+import math
+from processing.core.GeoAlgorithm import GeoAlgorithm
+from processing.core.parameters import ParameterMultipleInput, ParameterExtent, ParameterString, ParameterRaster, ParameterNumber
+from processing.core.outputs import OutputRaster
+from processing.tools import dataobjects
+from processing.algs.gdal.GdalUtils import GdalUtils
+from qgis.core import QgsRectangle
+from qgis.analysis import QgsRasterCalculator, QgsRasterCalculatorEntry
+from processing.core.GeoAlgorithmExecutionException import GeoAlgorithmExecutionException
+from processing.algs.qgis.ui.RasterCalculatorWidgets import LayersListWidgetWrapper, ExpressionWidgetWrapper
+
+
+class RasterCalculator(GeoAlgorithm):
+
+    LAYERS = 'LAYERS'
+    EXTENT = 'EXTENT'
+    CELLSIZE = 'CELLSIZE'
+    EXPRESSION = 'EXPRESSION'
+    OUTPUT = 'OUTPUT'
+
+    def defineCharacteristics(self):
+        self.name, self.i18n_name = self.trAlgorithm('Raster calculator')
+        self.group, self.i18n_group = self.trAlgorithm('Raster')
+
+        self.addParameter(ParameterMultipleInput(self.LAYERS,
+                                                 self.tr('Input layers'),
+                                                 datatype=dataobjects.TYPE_RASTER,
+                                                 optional=True,
+                                                 metadata={'widget_wrapper': LayersListWidgetWrapper}))
+
+        class ParameterExpression(ParameterString):
```

Comment:
I was not aware of that parameter...it seems you added it after I started working on this :-) I will change the name of my parameter, no problem